lib/kexec.nix

```diff
@@ -12,6 +12,40 @@ in {
     "${clever-tests}/kexec/kexec.nix"
     "${clever-tests}/kexec/justdoit.nix"
   ];
+
+  system.build = rec {
+    kexec_tarball = pkgs.callPackage <nixpkgs/nixos/lib/make-system-tarball.nix> {
+      storeContents = [
+        { object = config.system.build.kexec_script; symlink = "/kexec_nixos"; }
+      ];
+      contents = [];
+    };
+
+    kexec_tarball_self_extract_script = pkgs.writeTextFile {
+      executable = true;
+      name = "kexec-nixos";
+      text = ''
+        #!/bin/sh
+        ARCHIVE=`awk '/^__ARCHIVE_BELOW__/ { print NR + 1; exit 0; }' $0`
+
+        tail -n+$ARCHIVE $0 | tar xJ -C /
+        /kexec_nixos
+
+        exit 0
```

Comment:
`exit 1` or do we reach this line in the successful case?
Also adding `set -eu` would make sense to fail fast.